### ISSUE TITLE
Make DefaultHttpClientFactory not depend on MS DI

### ISSUE THREAD

**Issue by wasabii** [aspnet/HttpClientFactory#148]
Seems like all this logic would be nice to use with using the MS DI stuff. The only reason IServiceProvider is injected into it is for some sort of internal testing.

I'd like to just register it with Autofac. I don't use MS DI.

**Comment by davidfowl** [comment]
See this thread 
https://github.com/aspnet/HttpClientFactory/issues/55

In short, we’re not going to do that becuse the entire configuration API is built on it. You can build up your client then resolve the factory for use in autofac. Another approach would be to use the autofac adapter if you want to replace the container. See 
https://github.com/aspnet/HttpClientFactory/issues/127 for more details .

**Comment by davidfowl** [comment]
You don’t have to copy anything if you don’t want to just pretend like the di container is a configuration API and ignore it. Add the resulting imstance to autofac

**Comment by wasabii** [comment]
Can't. It pulls in too much stuff. Logging, etc.

**Comment by davidfowl** [comment]
So just so I'm clear on what is being asked for here. You want the piece of code that rotates handlers and nothing else in a separate package. Is that right?

**Comment by valeriob** [comment]
Yes !
Rotation and cleanup, just give an hook to create a new one when needed based on context (let us pass some parameter or a function to the CreateHttpClient to configure the factory).
One thing is not clear to me, who handles the connection pooling ? HttpClient or MessageHandler ?

**Comment by wasabii** [comment]
Yeah, I don't know how else to put this. This is a trivial library. I should be able to just new up a HttpClientFactory, stick it in a static. Or my own container. Or whatever.

Instead it's basically pegged to a specific DI implementation. I mean, that's a commonly known Bad Thing To Do. I'm not even sure how to further expand on it.

**Comment by davidfowl** [comment]
How many features did you remove along the way? I think it’s great you made it fit your needs but I’d love to know what you did anyways.

**Comment by wasabii** [comment]
Pretty much just took `DefaultHttpClientFactory` and removed the direct usage of `IServiceProvider` and such. `Func<HttpMessageHanderBuilder>` became a constructor arg.

Left the `IOptions` stuff.

No need for the Typed stuff.

**Comment by valeriob** [comment]
@wasabii did you add the possibility to pass parameter to the httpclient/handler factory ?
